precompiles/did/src/lib.rs

```diff
@@ -25,13 +25,14 @@ mod tests;
 #[precompile_utils::generate_function_selector]
 #[derive(Debug, PartialEq)]
 pub enum Action {
-	CreateDID = "createDID(address,address,uint8[],bytes[])",
-	CreateDIDOptional = "createDID(address,address,address,uint8[],bytes)",
-	RemoveDID = "removeDID(address)",
-	AddDIDServices = "addDidServices(address,uint8[],bytes[])",
-	RemoveDIDServices = "removeDidServices(address,bytes32[])",
-	IssueCredentials = "issueCredentials(address,address,bytes32,bytes)",
-	RevokeCredentials = "revokeCredentials(address,bytes32)",
+	CreateDID = "createDid(address,address,(bool,address),(uint8,string)[])",
+	UpdateDID =
+		"updateDid(address,(bool,address),(bool,address),(bool,address),(bool,(uint8,string)[]))",
+	RemoveDID = "removeDid(address)",
+	AddDIDServices = "addDidServices(address,(uint8,string)[])",
+	RemoveDIDServices = "removeDidServices(address,string[])",
```

Comment:
```suggestion
	RemoveDIDServices = "removeDidServices(address,bytes[])",
```

This one is actually a hash. It is the storage key of the services that should be removed.

Reply:
True! Fixed